### PR TITLE
fix type infer with UseFormHandleSubmit

### LINKED ISSUE #7404: issue: handleSubmit does not reject functions with invalid signature
### Version Number

7.4.1

### Codesandbox/Expo snack

https://codesandbox.io/s/eager-sun-jt4df?file=/src/App.tsx

### Steps to reproduce

1. useUseForm with some default values so it can infer the expected types
2. provide a function to handleSubmit that does not match the expected signature
3. typescript does not complain

For more details take a look to the provided sandbox

### Expected behaviour

I expect handleSubmit to not accept onSubmit, as it doesn't match the expected signature. A typescript error should be raised.

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by bluebill1049**
fix `UseFormHandleSubmit` with correct form values infer

### CODE REVIEW

File: src/types/form.ts [hunk 1]
```diff
@@ -261,7 +259,7 @@ export type UseFormResetField<TFieldValues extends FieldValues> = <
   }>,
 ) => void;
 
-export type UseFormReset<TFieldValues extends FieldValues> = (
+export type UseFormReset<TFieldValues extends FieldValues> = <_>(
```
Comment (kotarella1110) [inline src/types/form.ts:262]: @bluebill1049 Why did you change this?

File: src/__tests__/types/form.test-d.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      defaultValues: {
+        test: '',
+        test1: 2,
+      },
+    });
+
+    handleSubmit((data) => expectType<{ test: string; test1: number }>(data));
+  }
+
+  /** it should infer the correct defaultValues */ {
```
Comment (kotarella1110) [inline src/__tests__/types/form.test-d.ts:18]: I think we should change this title 👀

File: src/types/form.ts (hunk 1 above)
Comment (bluebill1049) [reply src/types/form.ts:262]: @kotarella1110 is there to avoid breaking change.

Comment (felixschorer) [reply src/types/form.ts:262]: @bluebill1049 that's `UseFormReset` 😅
`<_>` should be on `UseFormHandleSubmit`

Comment (bluebill1049) [reply src/types/form.ts:262]: oooops...

Comment (kotarella1110) [reply src/types/form.ts]: @bluebill1049 @felixschorer Thank you for explaining 👍🏻

**Comment by kotarella1110** [comment]
@bluebill1049 I think the type test you added will pass even with the type before you changed it.
Does the test you added have anything to do with #7404?

**Comment by bluebill1049** [comment]
not specific to the issue with the tests, just try to add more type test coverage in general. let me know if you would like to add a test case for the issue itself.

**Comment by bluebill1049** [comment]
@kotarella1110 are you happy to get this one in?

**Comment by bluebill1049** [comment]
> @kotarella1110 are you happy to get this one in?

I hope to merge this one today if you are busy. :)

**Comment by kotarella1110** [comment]
@bluebill1049 Sorry for the late reply. When I have time I will add the type test 👍🏻

**Pull request merged**